audience.go

```diff
@@ -10,15 +10,12 @@ import (
 )
 
 func (s *Snitch) addAudience(ctx context.Context, aud *protos.Audience) {
-	// Don't need to add twice
-	if s.seenAudience(ctx, aud) {
```

Comment:
Don't think this is needed - both seenAudience and addAudience are doing `if s.audiences == nil` checks and overwriting the map is non-expensive.

Reply:
It's needed because audiences aren't required at initialization and `Process()` must be able to announce new audiences. I do the seen check to avoid repetitive gRPC calls to `NewAudience()` every time `Process()` is called

Reply:
I have no idea why I didn't think about it actually doing outbound gRPC - every single time. I was just thinking about it re: map write.

Will add back in.

Looks like this change has both the audience stuff + reconnect. It's because I was working off of this branch for the other one (I think?)